### PR TITLE
MCR-68 Resolve saving and loading OSError on Windows

### LINKED ISSUE #68: OSError when auto-saving on Windows
```
Traceback (most recent call last):
  File "C:\Users\james\microcosm\game.py", line 255, in on_update
    if self.end_turn():
  File "C:\Users\james\microcosm\game.py", line 635, in end_turn
    self.save_game(auto=True)
  File "C:\Users\james\microcosm\game.py", line 824, in save_game
    with open(save_name, "w", encoding="utf-8") as save_file:
OSError: [Errno 22] Invalid argument: 'saves/autosave-2022-10-01T12:47:13.json'
PS C:\Users\james\microcosm> Traceback (most recent call last):                           File "C:\Users\james\microcosm\game.py", line 255, in on_update                                                         if self.end_turn():                                      File "C:\Users\james\microcosm\game.py", line 635, in end_turn                                                          self.save_game(auto=True)                                File "C:\Users\james\microcosm\game.py", line 824, in save_game                                                         with open(save_name, "w", encoding="utf-8") as save_file:                                                      
   OSError: [Errno 22] Invalid argument: 'saves/autosave-2022-10-01T12:47:13.json'
```

### PR DESCRIPTION
**Pull request by jamesgiu**
Tested on Windows 10 and Ubuntu, hopefully haven't broken anything. Let me know if anything else is required!

Closes #68

### CODE REVIEW

File: game.py [hunk 1]
```diff
@@ -821,10 +827,15 @@ def save_game(self, auto: bool = False):
...
         # Only maintain 3 autosaves at a time, delete the oldest if we already have 3 before saving the next.
-        if auto and len(autosaves := list(filter(lambda fn: fn.startswith("auto"), os.listdir("saves/")))) == 3:
+        if auto and len(
+                autosaves := list(filter(lambda fn: fn.startswith(AUTOSAVE_PREFIX), os.listdir(SAVES_DIR)))) == 3:
             autosaves.sort()
-            os.remove(f"saves/{autosaves[0]}")
-        save_name = f"saves/{'auto' if auto else ''}save-{datetime.datetime.now().isoformat(timespec='seconds')}.json"
+            os.remove(os.path.join(SAVES_DIR, {autosaves[0]}))
+        save_name = os.path.join \
+            (SAVES_DIR,
```
Comment (ChrisNeedham24) [inline game.py:835]: Some funky formatting here, the open bracket needs to be immediately after the `join`

File: game.py [hunk 2]
```diff
@@ -821,10 +827,15 @@ def save_game(self, auto: bool = False):
         Saves the current game with the current timestamp as the file name.
         """
         # Only maintain 3 autosaves at a time, delete the oldest if we already have 3 before saving the next.
-        if auto and len(autosaves := list(filter(lambda fn: fn.startswith("auto"), os.listdir("saves/")))) == 3:
+        if auto and len(
+                autosaves := list(filter(lambda fn: fn.startswith(AUTOSAVE_PREFIX), os.listdir(SAVES_DIR)))) == 3:
             autosaves.sort()
-            os.remove(f"saves/{autosaves[0]}")
-        save_name = f"saves/{'auto' if auto else ''}save-{datetime.datetime.now().isoformat(timespec='seconds')}.json"
+            os.remove(os.path.join(SAVES_DIR, {autosaves[0]}))
```
Comment (ChrisNeedham24) [inline game.py:833]: Don't think the braces are needed here, they were previously required because it was in a format string.

Comment (ChrisNeedham24) [reply game.py:833]: Yeah this actually crashes the game for me 💀

**Review by ChrisNeedham24 — CHANGES_REQUESTED** [review]
See the other two threads for required changes.

File: game.py (hunk 2 above)
Comment (jamesgiu) [reply game.py:833]: Good call, I'm curious if it's actually that causing your crash or something else though. Despite the brackets it seemed to work on Win/Ubuntu. Will look into it though, maybe missed a commit removing these or something.

Comment (jamesgiu) [reply game.py:833]: I also don't think I tested the autosave remove part so that might explain it!

Comment (jamesgiu) [reply game.py]: Replicated it on my end now and pushed a fix. My bad!

File: game.py (hunk 1 above)
Comment (ChrisNeedham24) [reply game.py]: Fixed in [d1ad791](https://github.com/ChrisNeedham24/microcosm/pull/70/commits/d1ad79100f613f104d44f75c4c56ee66be9a933c)

File: game.py (hunk 2 above)
Comment (ChrisNeedham24) [reply game.py]: Fixed in [d1ad791](https://github.com/ChrisNeedham24/microcosm/pull/70/commits/d1ad79100f613f104d44f75c4c56ee66be9a933c)

**Review by ChrisNeedham24 — APPROVED** [review]
Everything addressed nicely and is working!

**Pull request merged**